src/renderer/reducers/onboarding.ts

```diff
@@ -0,0 +1,160 @@
+import { DeviceModel, DeviceModelId } from "@ledgerhq/devices";
+import { handleActions } from "redux-actions";
+import { getEnv } from "@ledgerhq/live-common/lib/env";
+import { State } from ".";
+
+type Step = {
+  name: string;
+  external?: boolean;
+  label?: string;
+  options: {
+    showBreadcrumb: boolean;
+    relaunchSkip?: boolean;
+    alreadyInitSkip?: boolean;
+  };
+};
+
+export enum UseCase {
+  setupDevice = "setup-device",
+  connectDevice = "connect-device",
+  recoveryPhrase = "recovery-phrase",
+}
+
+export type OnboardingState = {
+  stepIndex: number;
+  stepName: string; // TODO: specify that the string comes from Steps type
+  deviceModelId?: DeviceModelId;
+  steps: Step[];
+  genuine: {
+    isDeviceGenuine: boolean;
+    displayErrorScreen: boolean;
+  };
+  onboardingRelaunched?: boolean;
+  useCase?: UseCase;
+};
+
+// type TutorialState;
+
+const initialState: OnboardingState = {
+  stepIndex: 0,
+  stepName: getEnv("SKIP_ONBOARDING") ? "analytics" : "start",
+  deviceModelId: undefined,
+  genuine: {
+    isDeviceGenuine: false,
+    displayErrorScreen: false,
+  },
+  onboardingRelaunched: false,
+  steps: [
+    {
+      name: "start",
+      external: true,
+      options: {
+        showBreadcrumb: false,
+      },
+    },
+    {
+      name: "init",
+      external: true,
+      options: {
+        showBreadcrumb: false,
+      },
+    },
+    {
+      name: "noDevice",
+      external: true,
+      options: {
+        showBreadcrumb: false,
+      },
+    },
+    {
+      name: "selectDevice",
+      label: "onboarding.breadcrumb.selectDevice",
+      options: {
+        showBreadcrumb: true,
+      },
+    },
+    {
+      name: "selectPIN",
+      label: "onboarding.breadcrumb.selectPIN",
+      options: {
+        showBreadcrumb: true,
+        alreadyInitSkip: true,
+      },
+    },
+    {
+      name: "writeSeed",
+      label: "onboarding.breadcrumb.writeSeed",
+      options: {
+        showBreadcrumb: true,
+        alreadyInitSkip: true,
+      },
+    },
+    {
+      name: "genuineCheck",
+      label: "onboarding.genuineCheck.title",
+      options: {
+        showBreadcrumb: true,
+      },
+    },
+    {
+      name: "setPassword",
+      label: "onboarding.breadcrumb.setPassword",
+      options: {
+        showBreadcrumb: true,
+        relaunchSkip: true,
+      },
+    },
+    {
+      name: "analytics",
+      label: "onboarding.breadcrumb.analytics",
```

Comment:
do we need to keep all info of states within the redux? i would just keep a list of step Ids and match it to the actual data elsewhere in the components that use and read this state. :thinking:

Reply:
that's the case or it's just not a reviewed zone now